core/src/main/scala/kafka/controller/KafkaControllerZkUtils.scala

```
@@ -347,6 +354,18 @@ class KafkaControllerZkUtils(zookeeperClient: ZookeeperClient, isSecure: Boolean
     retryRequestUntilConnected(setDataRequest)
   }
 
+  def createSequentialPersistentPath(path: String, data: String = ""): String = {
+    val createRequest = CreateRequest(path, data.getBytes("UTF-8"), acls(path), CreateMode.PERSISTENT_SEQUENTIAL)
+    val createResponse = retryRequestUntilConnected(createRequest)
+    createResponse.path
```

We want to throw an exception if the rc in createResponse is not Code.OK.